File: ruby/raindrops/raindrops.rb
module Raindrops
  def self.convert(n)
    result = ''
    result << 'Pling' if (n % 3).zero?
    result << 'Plang' if (n % 5).zero?
    result << 'Plong' if (n % 7).zero?
    result.empty? ? n.to_s : result
  end
end

File: python/raindrops/raindrops.py
FACTORS_TO_MESSAGE = (
    (3, "Pling"),
    (5, "Plang"),
    (7, "Plong")
)

def raindrops(number):
    result = "".join(message for factor, message in FACTORS_TO_MESSAGE if number % factor == 0)
    return str(number) if len(result) == 0 else result
